util/src/main/java/io/kubernetes/client/util/ConfigBuilder.java

```diff
@@ -0,0 +1,285 @@
+		return keyMgrs;
+	}
+
+	public ConfigBuilder setKeyMgrs(KeyManager[] keyMgrs) {
+		this.keyMgrs = keyMgrs;
+		return this;
+	}
+
+	public boolean isVerifyingSsl() {
+		return verifyingSsl;
+	}
+
+	public ConfigBuilder setVerifyingSsl(boolean verifyingSsl) {
+		this.verifyingSsl = verifyingSsl;
+		return this;
+	}
+
+	public boolean isDefaultClientMode() {
+		return defaultClientMode;
+	}
+
+	public ConfigBuilder setDefaultClientMode(boolean defaultClientMode) {
+		this.defaultClientMode = defaultClientMode;
+		return this;
+	}
+
+	public String getApiKeyPrefix() {
+		return apiKeyPrefix;
+	}
+
+	public ConfigBuilder setApiKeyPrefix(String apiKeyPrefix) {
+		this.apiKeyPrefix = apiKeyPrefix;
+		return this;
+	}
+
+	public ApiClient build()  {
+
+		ApiClient client = new ApiClient();
+
+		if( kubeConfig !=null) {
+			client = Config.fromConfig(kubeConfig);
+			return client;
+		}
+
+		if(defaultKubeConfigMode == true) {
+			try {
+				client = Config.fromConfig(KubeConfig.loadDefaultKubeConfig());
+			} catch (FileNotFoundException e) {
+				e.printStackTrace();
+			}
+			return client;
+		}
+
+		if(clusterMode == true) {
+			try {
+				client = Config.fromCluster();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return client;
+		}
+
+		if(defaultClientMode ==true ) {
+			try {
+				client = Config.defaultClient();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return client;
+
+		}
+
+		if (basePath != null ) {
+			if(basePath.endsWith("/")) {
+				basePath = basePath.substring(0, basePath.length() - 1);
+			}
+			client.setBasePath(basePath)
+			.setVerifyingSsl(verifyingSsl);
+		}
+
+		else {
+			try {
+				throw new Exception("set kubernetes URL. example: http://localhost");
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+
+		if(keyMgrs != null) {
+			client.setKeyManagers(keyMgrs);
+		}
+
+		if(userName != null){
+			client.setUsername(userName);
+		}
+
+		if(password != null){
+			client.setPassword(password);
+		}
+
+		if(( userName != null )&&(password != null)) {
+			final String usernameAndPassword = userName + ":" + password;
+			client.setApiKeyPrefix("Basic");
+			client.setApiKey(ByteString.of(usernameAndPassword.getBytes(Charset.forName("ISO-8859-1"))).base64());
+		}
+
+		if(accessToken != null) {
+			if (apiKeyPrefix == null){
+				client.setApiKeyPrefix("Bearer");
+			}
+			client.setAccessToken(accessToken);
+		}
+
+		if(apiKeyPrefix != null) {
+			client.setApiKeyPrefix(apiKeyPrefix);
+		}
+
+		if(apiKey != null) {
+			if (apiKeyPrefix == null){
+				client.setApiKeyPrefix("Bearer");
+			}
+			client.setApiKey(apiKey);
+		}
+
+		if(sslCaCert != null) {
+			client.setSslCaCert(sslCaCert);
+		}
+		
+		if(verifyingSsl){
+			if((certificateAuthorityData != null) || (certificateAuthorityFile != null)){
+				try {
+					client.setSslCaCert(SSLUtils.getInputStreamFromDataOrFile(certificateAuthorityData, certificateAuthorityFile));
+				} catch (FileNotFoundException e) {
```

Comment:
I think you should load the cert when someone calls `setCertificateAuthority(...)`

And not throw/catch here.